tests/test_sdist.py

```diff
@@ -93,15 +66,15 @@ def test_read(archive_format, tmp_path):
         },
     )
 
-    metadata = sdist.SDist(filename).read()
+    metadata = sdist.SDist(str(filepath)).read()
     assert b"Metadata-Version: 1.1" in metadata
     assert b"Name: test" in metadata
     assert b"Version: 1.2.3" in metadata
 
 
 def test_missing_pkg_info(archive_format, tmp_path):
     """Raise an exception when sdist does not contain PKG-INFO."""
-    filename = build_archive(
+    filepath = build_archive(
```

Comment:
As an observation: we end up calling `str(filepath)` repeatedly, so maybe it's fine to have `build_archive` return a `str` instead of a `pathlib.Path`? 

(No strong preference, however.)

Reply:
The reason for the change is explained in the commit message. In subsequent patches I'll use the `Path` object instead of the string. I've a few more cleanups ready to submit but I split them in smaller PRs for easy of review.